File: erlang/hello.erl
%% Run it:
%% erl
%% c(hello).
%% hello:greet().

%% erlc hello.erl && erl -noshell -s hello greet -s init stop

-module(hello).
-export([greeter/1, greet/0]).


greet() ->
    greeter("DevFest").

greeter(Name) ->
    io:fwrite("~s ~s ~n", ["Hello", Name]).

File: scala/hello.scala

/**
  * Execute from the command line like:
  *
  *   scala hello.scala [name]
  *
  */

// "Greeter" is a singleton object instance
object Greeter {
  def greet(name: String) {
    println(s"Hello $name!")
  }
}

val name = if(args.size > 0) args(0) else "DevFest"

Greeter.greet(name)

